src/module.js

```diff
@@ -100,27 +98,31 @@ function processModule( name, options, executeNow, modifiers = {} ) {
 	}
 }
 
+let focused = false;
+
 export default function module( name, options, executeNow ) {
 	if ( focused && !isParentModuleInQueue() ) {
+		config.currentModule.closed = true;
 		return;
 	}
 
 	processModule( name, options, executeNow );
 }
 
-module.only = function() {
+module.only = function( ...args ) {
 	if ( !focused ) {
```

Comment:
Based on how `test.only` is written, I would expect `focused = true;` to be set in this block instead of after calling `processModule()`.

If changing this would not affect the issue that this patch addresses, then feel free to leave it as-is and we can look at it later (to keep this patch clean).

Reply:
This is agnostic to _flat_ modules, but here's a tangible test to study with _nested_ modules:
```
QUnit.module.only( "Module A", () => {
    QUnit.module( "Module B", () => {
        QUnit.test( "Test A", function( assert ) {
            assert.true( true );
        } );
    } );
    QUnit.module.skip( "Module C", () => {
        QUnit.test( "Test B", function( assert ) {
            assert.true( true );
        } );
    } );
} );
```

With setting `focused = true` where it is today, `Test A` is run and `Test B` is skipped:
```
ok 1 Module A > Module B > Test A
ok 2 # SKIP Module A > Module C > Test B
1..2
# pass 1
# skip 1
# todo 0
# fail 0
```
Pushing the `focused = true` to before `processModule()` gives just `Test A`, and `Test B` is never even registered:
```
ok 1 Module A > Module B > Test A
1..1
# pass 1
# skip 0
# todo 0
# fail 0
```

The behavior is maybe debatable, but for the scope of this PR. I'd prefer we keep it as-is for compatibility, and maybe consider separately in the future.